apps/src/lib/proto/types.rs

```diff
@@ -0,0 +1,329 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+use prost::Message;
+use thiserror::Error;
+
+use super::generated::{services, types};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Error decoding a transaction from bytes: {0}")]
+    TxDecodingError(prost::DecodeError),
+    #[error("Error decoding an IntentGossipMessage from bytes: {0}")]
+    IntentDecodingError(prost::DecodeError),
+    #[error("Error decoding an DkgGossipMessage from bytes: {0}")]
+    DkgDecodingError(prost::DecodeError),
+    #[error("Intent is empty")]
+    NoIntentError,
+    #[error("Dkg is empty")]
+    NoDkgError,
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
+pub struct Tx {
+    inner: types::Tx,
+}
```

Comment:
As it's only a wrapper you could use instead : 
```rust
pub struct Tx (types::Tx);
```
I'm not sure if it's preferable, both are acceptable I feel

Reply:
I know there's a some of repetition involved in this, but I'd be in favor of actually writing the types here by hand, because proto3 puts optional on many fields that are actually not optional. E.g. for the `Tx`:

```rust
pub struct Tx {
    pub code: Vec<u8>,
    pub data: Option<Vec<u8>>,
    pub timestamp: Timestamp,
}
```

The `data` is declared as optional, but `timestamp` should be required in a valid `Tx`. The validation layer would then make the data nicer for consumers of these types

Reply:
I aggree with @tzemanovic, It's better to hide protobuf option

Reply:
Thank you! I set fields for each type.